src/Builder/Select/ExpressionBuilder.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Solarium package.
+ *
+ * For the full copyright and license information, please view the COPYING
+ * file that was distributed with this source code.
+ */
+
+namespace Solarium\Builder\Select;
+
+use Solarium\Builder\Comparison;
+use Solarium\Builder\CompositeComparison;
+
+/**
+ * Expression Builder.
+ *
+ * @author wicliff <wicliff.wolda@gmail.com>
+ */
+class ExpressionBuilder
+{
+    /**
+     * @param null $x
+     *
+     * @throws \Solarium\Exception\RuntimeException
+     *
+     * @return \Solarium\Builder\CompositeComparison
+     */
+    public function andX($x = null): CompositeComparison
+    {
+        return new CompositeComparison(CompositeComparison::TYPE_AND, \func_get_args());
+    }
```

Comment:
The [`...` operator](https://www.php.net/manual/en/functions.arguments.php#functions.variable-arg-list) makes this typehintable and thus more readable.

```suggestion
    /**
     * @param Comparison ...$comparison
     *
     * @throws \Solarium\Exception\RuntimeException
     *
     * @return \Solarium\Builder\CompositeComparison
     */
    public function andX(Comparison ...$comparison): CompositeComparison
    {
        return new CompositeComparison(CompositeComparison::TYPE_AND, $comparison);
    }
```

Reply:
i've been working quite a bit with the [FunctionBuilder](https://github.com/solariumphp/solarium/blob/master/src/Builder/Analytics/FunctionBuilder.php) recently and one of the benefits of not typehinting the arguments is that you can pass a string as argument for the edge cases you run into.

for instance creating this function [``filter(mult(price,quantity),or(cats:equal(category,_)))``](https://lucene.apache.org/solr/guide/analytics.html#for-each-functions) would translate into
```php
$expression = FunctionBuilder::create()
            ->where($expr->filter($expr->mult('price', 'quantity'), $expr->or('cats:equal(category,_)')))
        ;
```
as ``$expr->equals`` would render a different expression, it cannot be used to generate the desired result.

as this is a helper class, i'd personally prefer to make it not too strict; what do you think?